app/src/main/java/com/amaze/filemanager/filesystem/UsbOtgSingleton.java

```diff
@@ -0,0 +1,39 @@
+package com.amaze.filemanager.filesystem;
+
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+public class UsbOtgSingleton {
+    private static UsbOtgSingleton instance = null;
+
+    public static UsbOtgSingleton getInstance() {
+        if(instance == null) instance = new UsbOtgSingleton();
+        return instance;
+    }
+
+    private Uri usbOtgRoot = null;
+    /**
+     * Indicates whether last app exit was for setting {@link #usbOtgRoot} or not
+     */
+    private boolean hasRootBeenRequested = false;
+
+    private UsbOtgSingleton() { }
+
+    public void setUsbOtgRoot(@Nullable Uri root) {
+        usbOtgRoot = root;
+    }
+
+    public @Nullable Uri getUsbOtgRoot() {
+        return usbOtgRoot;
+    }
+
+    public void setHasRootBeenRequested(boolean hasRootBeenRequested) {
```

Comment:
Sorry for the late reply, have been without much time.

> I think this method is redundant. There's no actual use of this, we can simply perform this same action by checking if the Uri object is null or not.

Not true, if `usbOtgRoot` is null this means it hasn't been set. `hasRootBeenRequested` means that it will be set in the future. Not every time `usbOtgRoot` is nullified it will be set in the future, so we cannot know if it will be set in the future from checking for null.

I think that before I changed it this information was saved to a SharedPreference as three states (null, default, and a valid uri).

> This is the only method that's preventing everything mentioned in this class from going to OTGUtil instead. Hence I think this class itself is redundant.

It is redundant, as every other singleton, but having static variables everywhere is not a good idea, because any object instance can change it directly for every other object (idem to not having global variables in C++ applies here or public fields to objects in Java).